src/sinks/tcp.rs

```diff
@@ -227,3 +403,17 @@ fn encode_event(event: Event, encoding: &Option<Encoding>) -> Result<Bytes, ()>
         Bytes::from(b)
     })
 }
+
+trait FramedConnection {
```

Comment:
I believe we should try to abstract around Framed<IO> instead of a FramedConnection.

Reply:
I'm not sure what you are describing here. It might be moot, as this goes away if I can get the `MaybeTlsStream` change implemented.